OpenImis.ePayment/Escape/Payment/Data/ImisPayment.cs

```diff
@@ -451,7 +452,47 @@ Amount DECIMAL(18, 2)
 
         }
 
-        
+        public int ControlNumbersToBeRequested(string productCode)
+        {
+            var sSQL = @";WITH TotalProductUsage
```

Comment:
Well, when it comes to resources, table variables are definitely more resource expensive than CTE because table variable resides in tempdb and consumes space in file as well as in memory. Also, because they are treated as real tables they also come with all the characteristics  like indexes, constraints and all. Moreover, we use table variables if we need to perform more actions on data. In our case we are just reading.

Coming back to a numeric variable, we could use it too. It is just a preference and readability, it has no impact on performance.